src/Components/Login/Login.tsx

```diff
@@ -19,56 +13,61 @@ const ALL_USERS = gql `
   }
 `
 
-const ALL_BOOKS = gql `
-  query AllBooks {
-    books {
-      id
-      imageUrl
-    }
-  }
-`
-
 const Login = () => {
   const [email, setEmail] = useState('')
   const [username, setUsername] = useState('')
   const [userLocation, setUserLocation] = useState('')
   const [activeAccount, setAccount] = useState(true)
-  const [allUsers, setUsers] = useState([{userName: "Josh", emailAddress: "test", location: "Denver"}])
+  const [loginError, setLoginError] = useState(false)
```

Comment:
You have both `loginError` state, as well as `errorMessage` state. I understand that one's purpose is for determining if there is an error, and the other contains actually error information.

I'd recommend "combining" these two into one state. You can get rid of `loginError` and keep `errorMessage` and accomplish the same things.

If `errorMessage` has any value in it, you can assume there was an error. If it doesn't, there is no error. Therefore line 70 can change to something like:
```TS
{!!errorMessage.length && <h3>errorMessage</h3>}
```

Anytime you update state in react, you're causing a rerender of that component -> down the tree. Most of these rerenders are pretty fast, and ultimately harmless but it's good to be aware of them all the same and try and mitigate the number of rerenders you do.

By having two separate states for this, anytime you update one of them, you rerender. So lines 45,46 and 49,50 and 53,54. In those cases you set loginError to true, you rerender, then resume at that function and set errorMessage to a value, and rerender again.